### PR TITLE
Added dropbox storage

### PR DESCRIPTION
**Pull request by ZuluPro**
- Supports Dropbox oAuth API
- Added with unittests
- Maybe you want some doc in this pull request ?

Further:
- Support auth with API key and secret
- Ensure locations are great

### CODE REVIEW

**Comment by ZuluPro** [comment]
I had some troubles with dropbox library and Python 3.
The library installed from SDK doesn't seem to be up to date and doesn't work with Py3.

I instead use their official repo: https://github.com/dropbox/dropbox-sdk-python
with `pip install https://github.com/dropbox/dropbox-sdk-python/archive/master.zip`

**Comment by ZuluPro** [comment]
Any news ?

**Comment by jschneier** [comment]
The code looks great.

File: storages/backends/dropbox.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# FTP storage class for Django pluggable storage system.
```
Comment (jschneier) [inline storages/backends/dropbox.py]: let's remove the FTP stuff :)

File: setup.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ def read(filename):
         'Django>=1.6.2',
         'pytest',
         'mock',
-        'boto>=2.32.0'
+        'boto>=2.32.0',
+        'dropbox>=2.2.0'
```
Comment (jschneier) [inline setup.py]: how does this relate to the tarball from master that we are installing down below?

**Comment by ZuluPro** [comment]
All is done!

File: setup.py (hunk 2 above)
Comment (ZuluPro) [inline setup.py]: I have correct this by add a ``requirements-test.txt`` and this line in ``setup.py``:

```
...
tests_require=readlines('requirements-tests.txt'),
...
```

**Comment by ZuluPro** [comment]
I don't know why Travis don't launch test for my pull request, but it successes here: https://travis-ci.org/ZuluPro/django-storages/builds/79368054

**Comment by ZuluPro** [comment]
@jschneier Any news ?

**Pull request merged**